pios_video: correct PIOS_VIDEO_INPUT_FILTER

mlyle opened the pull request:
bad news: i'm a moron
good news: it's very likely this is a cause of #1681 

Basically, #1640 had two changes:

1. stop using a line counter with various dubious logic [this one is good]
2. attempt, on Seppuku, to use an edge filter on hsync for robustness when colorburst toggling gets through on the hsync line.   I messed this one up, and programmed the bits to the wrong register.  The net result was it had no effect.

I've tested this on PAL and it is good.  Feedback from people with NTSC problems (@litterbug ?   @pug398 ?)  would be much appreciated

mlyle committed:
pios_video: correct PIOS_VIDEO_INPUT_FILTER

pug398 commented:
jenkins, build artifacts please

mlyle commented:
sorry, i forgot to ask for a build :/  CI was broken until just now anyways heh.

jenkins, build artifacts please

pug398 commented:
Works good for me on the bench with no sensor warnings/resets. The only thing I can force it to do on occasion is to lose vertical lock and that is because my camera output changes sync level with high contrast change. Don't think there is much you can do in code about that.

mlyle commented:
@pug398 --- thanks for the test.  I'm glad it works for you.  Just one small point of clarification-- is there any difference that you saw between this and the previous changeset/current next)? 

Verris from IRC indicates he's going to be able to test later, so it'll be interesting to see if it cures his problem.

mlyle committed:
pios_video: less aggressive filter, use diff interrupt

mlyle committed:
pios_video: thought of one more trick

Since we're using Update, we can limit the rate at which we get
interrupts further, by taking about as long to cycle through the
counts as one line during line-counting.

mlyle commented:
Two changes, now.  1) Saw a small video quality regression, so made the filter less aggressive (but still low enough frequency to reject colorburst) 2) Verris's problem wasn't fixed, so attempt to manage rate of line counter interrupt by ensuring the timer takes *nearly* as long as a line to wrap/update.

Jenkins, build artifacts please

pug398 commented:
@mlyle actually the fix before this one took care of most of my issues and there wasn't any real difference. I Will try the latest incarnation later today.

mlyle commented:
This still doesn't fix Verris's problem.  It's a good change, though.

mlyle committed:
pios_video: detect modes more robustly

Also prevent from accessing invalid pointers when wacky things happen.

mlyle commented:
jenkins, build artifacts please

mlyle commented:
@litterbugs curious if this fixes your problem (&verris's)

LitterBugs commented:
@mlyle This does appear to have fixed the OSD startup issue.

LitterBugs commented:
One other thing I have noticed about the Seppuku OSD config is that the default NTSC Black level of 20 appears to be set too low.  The black background around the OSD elements of the screen seems to unsettle the video sync as a whole.  Depending on what display I am using (LCD Panel or Goggles) this shows up differently one Tears and ghosts, the other the Vsync jumps on the whole video image.  Bumping it up to 35 like PAL is set stabilizes the signal.

LitterBugs commented:
One last thought.  With all the recent changes, it is no longer necessary to use the y offset.  Image appears centered on both my ground station and goggles

mlyle commented:
@litterbugs -- thanks for the test and the feedback.  Unfortunately it's hard with current OSD infrastructure to set per-board levels, though I may be able to provide a small nudge that way for Seppuku.

LitterBugs commented:
@mlyle No worries, it is easy enough to adjust.  Couldn't remember if I'd said anything about that previously.  Remembered it when setting everything back to defaults for testing the other night.  I can't remember if it is just the Pre-release board or if the final board does it too.  Could also be a combination of the board and VTX.  I haven't hooked my released board up through a VTX yet.

pug398 commented:
@LitterBugs I run NTSC black level from 35 to 50 to get rid of various artifacts so it is not just you. I do have to set y offset in some cases still.

mlyle commented:
#1692 offsets the black--- and white--- levels +12.

tracernz reviewed (approved):
:grin:
